src/lib/server.ts

```diff
@@ -434,7 +435,9 @@ export class ZwavejsServer extends EventEmitter {
     this.sockets.on("hard reset", async () => {
       this.driver.once("driver ready", () => this.start());
       await this.driver.hardReset();
-      await this.destroy();
+      if (this.destroyServerOnHardReset) {
```

Comment:
added this logic because `zwave-js-ui` will destroy the server on its own as part of the restart so best to let it do its thing instead of destroying the server twice.